myhealth application crash

FATAL EXCEPTION: main
                                                                                                    Process: org.ole.planet.myplanet, PID: 19736
                                                                                                    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.ole.planet.myplanet/org.ole.planet.myplanet.ui.myhealth.AddExaminationActivity}: java.util.IllegalFormatConversionException: d != java.lang.Float
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4048)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4235)
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:112)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeNonLifecycleItem(TransactionExecutor.java:174)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:109)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:81)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2636)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:107)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232)
                                                                                                    	at android.os.Looper.loop(Looper.java:317)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8705)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:886)
                                                                                                    Caused by: java.util.IllegalFormatConversionException: d != java.lang.Float
                                                                                                    	at java.util.Formatter$FormatSpecifier.failConversion(Formatter.java:4608)
                                                                                                    	at java.util.Formatter$FormatSpecifier.printInteger(Formatter.java:3123)
                                                                                                    	at java.util.Formatter$FormatSpecifier.print(Formatter.java:3078)
                                                                                                    	at java.util.Formatter.format(Formatter.java:2704)
                                                                                                    	at java.util.Formatter.format(Formatter.java:2640)
                                                                                                    	at java.lang.String.format(String.java:4068)
                                                                                                    	at android.content.res.Resources.getString(Resources.java:584)
                                                                                                    	at android.content.Context.getString(Context.java:962)
                                                                                                    	at org.ole.planet.myplanet.ui.myhealth.AddExaminationActivity.initExamination(AddExaminationActivity.kt:100)
                                                                                                    	at org.ole.planet.myplanet.ui.myhealth.AddExaminationActivity.onCreate(AddExaminationActivity.kt:86)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:9002)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:8980)
                                                                                                    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1526)
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4030)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4235) 
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:112) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeNonLifecycleItem(TransactionExecutor.java:174) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:109) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:81) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2636) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:107) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232) 
                                                                                                    	at android.os.Looper.loop(Looper.java:317) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8705) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:886)


https://github.com/user-attachments/assets/7d4d816c-ba9d-4e20-be4b-629e2589ef3e

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -10,8 +10,8 @@ android {
         applicationId "org.ole.planet.myplanet"
         minSdkVersion 26
         targetSdkVersion 34
-        versionCode 2270
-        versionName "0.22.70"
+        versionCode 2279
+        versionName "0.22.79"
         ndkVersion '21.3.6528147'
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         vectorDrawables.useSupportLibrary = true

--- a/app/src/main/java/org/ole/planet/myplanet/ui/myhealth/AddExaminationActivity.kt
+++ b/app/src/main/java/org/ole/planet/myplanet/ui/myhealth/AddExaminationActivity.kt
@@ -97,11 +97,11 @@ class AddExaminationActivity : AppCompatActivity(), CompoundButton.OnCheckedChan
     private fun initExamination() {
         if (intent.hasExtra("id")) {
             examination = mRealm.where(RealmMyHealthPojo::class.java).equalTo("_id", intent.getStringExtra("id")).findFirst()!!
-            activityAddExaminationBinding.etTemperature.setText(getString(R.string.number_placeholder, examination?.temperature))
+            activityAddExaminationBinding.etTemperature.setText(getString(R.string.float_placeholder, examination?.temperature))
             activityAddExaminationBinding.etPulseRate.setText(getString(R.string.number_placeholder, examination?.pulse))
             activityAddExaminationBinding.etBloodpressure.setText(getString(R.string.message_placeholder, examination?.bp))
-            activityAddExaminationBinding.etHeight.setText(getString(R.string.number_placeholder, examination?.height))
-            activityAddExaminationBinding.etWeight.setText(getString(R.string.number_placeholder, examination?.weight))
+            activityAddExaminationBinding.etHeight.setText(getString(R.string.float_placeholder, examination?.height))
+            activityAddExaminationBinding.etWeight.setText(getString(R.string.float_placeholder, examination?.weight))
             activityAddExaminationBinding.etVision.setText(examination?.vision)
             activityAddExaminationBinding.etHearing.setText(examination?.hearing)
             val encrypted = user?.let { examination?.getEncryptedDataAsJson(it) }

--- a/app/src/main/res/values-ar/strings.xml
+++ b/app/src/main/res/values-ar/strings.xml
@@ -1092,6 +1092,7 @@
     <string name="message_placeholder">%s</string>
     <string name="reports">التقارير</string>
     <string name="number_placeholder">%d</string>
+    <string name="float_placeholder">%.1f</string>
     <string name="date_range">%1$s إلى %2$s</string>
     <string name="string_range">%1$s - %2$s</string>
     <string name="team_financial_report">%s تقرير مالي لـ</string>

--- a/app/src/main/res/values-es/strings.xml
+++ b/app/src/main/res/values-es/strings.xml
@@ -1092,6 +1092,7 @@
     <string name="message_placeholder">%s</string>
     <string name="reports">informes</string>
     <string name="number_placeholder">%d</string>
+    <string name="float_placeholder">%.1f</string>
     <string name="date_range">%1$s a %2$s</string>
     <string name="string_range">%1$s - %2$s</string>
     <string name="team_financial_report">Informe financiero de %s</string>

--- a/app/src/main/res/values-fr/strings.xml
+++ b/app/src/main/res/values-fr/strings.xml
@@ -1092,6 +1092,7 @@
     <string name="message_placeholder">%s</string>
     <string name="reports">rapports</string>
     <string name="number_placeholder">%d</string>
+    <string name="float_placeholder">%.1f</string>
     <string name="date_range">%1$s à %2$s</string>
     <string name="string_range">%1$s - %2$s</string>
     <string name="team_financial_report">Rapport financier de %s</string>

--- a/app/src/main/res/values-ne/strings.xml
+++ b/app/src/main/res/values-ne/strings.xml
@@ -1092,6 +1092,7 @@
     <string name="message_placeholder">%s</string>
     <string name="reports">रिपोर्टहरू</string>
     <string name="number_placeholder">%d</string>
+    <string name="float_placeholder">%.1f</string>
     <string name="date_range">%1$s देखि %2$s</string>
     <string name="string_range">%1$s - %2$s</string>
     <string name="team_financial_report">%s वित्तीय रिपोर्ट</string>

--- a/app/src/main/res/values-so/strings.xml
+++ b/app/src/main/res/values-so/strings.xml
@@ -1092,6 +1092,7 @@
     <string name="message_placeholder">%s</string>
     <string name="reports">warbixino</string>
     <string name="number_placeholder">%d</string>
+    <string name="float_placeholder">%.1f</string>
     <string name="date_range">%1$s ilaa %2$s</string>
     <string name="string_range">%1$s - %2$s</string>
     <string name="team_financial_report">Warbixin lacagta ee %s</string>

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -1092,6 +1092,7 @@
     <string name="message_placeholder">%s</string>
     <string name="reports">Reports</string>
     <string name="number_placeholder">%d</string>
+    <string name="float_placeholder">%.1f</string>
     <string name="date_range">%1$s to %2$s</string>
     <string name="string_range">%1$s - %2$s</string>
     <string name="team_financial_report">%s Financial Report</string>

